Sources/ServiceLifecycle/CancellableContinuation.swift

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftServiceLifecycle open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftServiceLifecycle project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftServiceLifecycle project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A simple actor that provides a method that suspends forever until cancelled.
+actor CancellableContinuation {
+    /// The continuation.
+    private var continuation: CheckedContinuation<Void, Never>?
+
+    /// This method suspends forever until it is cancelled.
+    func run() async throws {
```

Comment:
Why does this throw?

Reply:
It is very normal for `run()` method implementations to throw. What NIO based servers often do in run is call `bind` which can throw. There are other things that users might want to do in here that can throw such as iterating a throwing  `AsyncSequence`.

Reply:
True but this is an explicit implementation that does not throw :)

Reply:
Ah sorry, misread where the comment is.